### PR TITLE
Working, Installable version of pyPeakError

### PR DESCRIPTION
**Pull request by deltarod**
Implements a C binding for PeakError. It can perform PeakError and PeakErrorChrom found [here](https://github.com/tdhock/PeakError/blob/master/R/PeakError.R). It can also summarize an error file 

@tdhock

### CODE REVIEW

File: PeakError/util.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        return 'false positive'
+    else:
+        return 'correct'
+
+
+def fnCalc(row):
+    return row['possible_tp'] - row['tp']
+
+
+def fnStatus(val):
```
Comment (tdhock) [inline PeakError/util.py]: fnStatus / fpStatus probably can be removed, I never really used them in R.

File: PeakError/util.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    if val > 0:
+        return 'false negative'
+    else:
+        return 'correct'
+
+
+def status(row):
```
Comment (tdhock) [inline PeakError/util.py:51]: status is useful in R for visualization purposes, may be good to keep

File: PeakError/util.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    if row['fn'] > 0:
+        return "false negative"
+    else:
```
Comment (tdhock) [inline PeakError/util.py]: you can remove else here and below, because return used below

File: src/interface.c [hunk 4]
```diff
@@ -1,53 +1,139 @@
...
-	error("overlapping regions");
+        PyErr_SetString(PyExc_ValueError,
+                        "overlapping regions");
     }
     if(status != 0){
-	error("error code %d", status);
+        return NULL;
     }
+
+    return Py_BuildValue("{s:N,s:N,s:N,s:N}",
```
Comment (tdhock) [inline src/interface.c:105]: all this interface code looks good

File: tests/test_PeakError.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    peaksdf = pd.read_csv(peaksPath, sep='\t', header=None, index_col=False)
+    peaksdf.columns = ['chrom', 'chromStart', 'chromEnd']
+
+    regionsPath = '%slabels.bed' % exampleDataPath
+    regionsdf = pd.read_csv(regionsPath, sep='\t', header=None, index_col=False)
+    regionsdf.columns = ['chrom', 'chromStart', 'chromEnd', 'annotation']
+
+    output = PeakError.error(peaksdf, regionsdf)
+
+    testfp = [1, 1, 1, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 0, 1, 0]
```
Comment (tdhock) [inline tests/test_PeakError.py:39]: these test cases are copied from https://github.com/tdhock/PeakError/blob/master/tests/testthat/test-examples.R ?

**Comment by tdhock** [comment]
overall looks good, next step is to think about how to use this in PeakLearner. Typically I would expect you run PeakError every time you get a new label/model, and then store the resulting fp/fn values in the PeakLearner database.

File: tests/test_PeakError.py (hunk 5 above)
Comment (deltarod) [reply tests/test_PeakError.py:39]: No, I noticed in https://github.com/tdhock/PeakError/blob/master/tests/script.R that you had issues using testthat so I didn't really look too much into those tests. I was already planning on writing some more tests, but getting it working was the main priority

Comment (tdhock) [reply tests/test_PeakError.py:39]: well anyway the tests pass in the R version so if you copied the C code it should work the same

Comment (deltarod) [reply tests/test_PeakError.py:39]: PeakError.c/h should be identical

**Pull request merged**